pkg/instancetype/instancetype.go

```diff
@@ -59,32 +59,35 @@ func GetRevisionName(vmName, resourceName string, resourceUID types.UID, resourc
 	return fmt.Sprintf("%s-%s-%s-%d", vmName, resourceName, resourceUID, resourceGeneration)
 }
 
-func CreateInstancetypeControllerRevision(vm *virtv1.VirtualMachine, revisionName string, instancetypeApiVersion string, instancetypeSpec *instancetypev1alpha1.VirtualMachineInstancetypeSpec) (*appsv1.ControllerRevision, error) {
+func CreateControllerRevision(vm *virtv1.VirtualMachine, revisionName string, object runtime.Object) *appsv1.ControllerRevision {
+	objCopy := object.DeepCopyObject()
 
-	instancetypeSpecPatch, err := json.Marshal(*instancetypeSpec)
+	// TypeMeta may be missing, so set it here.
+	gvks, _, err := generatedscheme.Scheme.ObjectKinds(objCopy)
 	if err != nil {
-		return nil, err
+		// Using panic because this is a programming error
+		panic(fmt.Errorf("could not get GroupVersionKind for object: %w", err))
```

Comment:
This really doesn't strike me as great practice even if as you say we can only get here through some dev failure at the moment. Do we really want to panic at all in `virt-controller` or `virt-api` ?

Reply:
@davidvossel do you know if it is ok to use `panic` here?

Reply:
in general, i typically see `panic()` used during initialization, when some initialization step fails and there's no alternative but to exit. So the panic is useful because it prevents either unrecoverable or undefined behavior.

I also occasionally see functions called `<DoSomething>OrDie()`, like fields.ParseSelectorOrDie() for example. That `OrDie` part is kind of signifying to the caller that this is either going to pass, or kill your process. I wouldn't say that's super common.

Since the functionality in question is occurring after initialization within the reconcile loop or during a subresource http request, I'd be inclined to handle it as an error rather than panic.

Reply:
Thank you for the explanation, I will change it to error return.